src/sync/parallel-page-fetcher.ts

```diff
@@ -31,15 +31,21 @@ export const fetchNextPagesInParallel = async (
 		},
 		(_, index) => singlePageFetchFactory(nextPageCursor + index)
 	);
-	const fetchedData = await Promise.all(tasks);
-	return fetchedData.reduce((prev, curr) => {
-		return {
-			edges: [...(prev.edges || []), ...(curr.edges || [])],
-			jiraPayload:
-				mergeJiraPayload(prev.jiraPayload, curr.jiraPayload)
-		};
-	}, {
+	const fetchedData = await Promise.allSettled(tasks);
```

Comment:
Why from `all` to `allSettled`? 
1. That might reduce performance and wastes rate limit quota
2. If only worry about the structure of the error, just a simple try/catch will do. 
3. The `if (curr.status === "fulfilled") {...} else {...throw curr.reason;}` I think reduce the code readability, compared to `try { fetchedData = await Promise.all()... } catch (e) { throw transform(e) }

But okay I don't see a big problem though.

Reply:
This makes tests more deteministic, otherwise there could be some calls happening while the test is already over.

Reply:
Also, mind you, this is only a error case when it might take longer time, which is a corner-case